File: posts/code/recruit-benkyokai-20210115/mcmc.py
import sys
from random import random

def pi_mcmc(n):
    t = 0
    for i in range(n):
        x, y = random(), random()
        if x**2 + y**2 <= 1:
            t += 1

    return 4*t / n

print(pi_mcmc(int(sys.argv[1])))

File: posts/code/julia-no-susume-20210115/mcmc.c
#include <stdlib.h>
#include <time.h>
#include <stdlib.h>

double pi_mcmc(long n) {
    srand((unsigned)time(NULL));
    int t = 0;
    for (long i = 1; i <= n; i++) {
        double x = (double)rand() / RAND_MAX;
        double y = (double)rand() / RAND_MAX;
        if (x * x + y * y <= (double) 1.0)
            t++;
    }
    return ((double) 4.0)*((double) t)/((double) n);
}

int main(int argc, char *argv[]) {
   long n = atol(argv[1]);
   printf("%f\n", pi_mcmc(n));
   return 0;
}
